src/x/parsers/metric_parsers.go

```diff
@@ -0,0 +1,51 @@
+package parsers
+
+import (
+	"bytes"
+
+	"github.com/m3db/m3/src/metrics/metric/id"
+	"github.com/m3db/m3/src/x/serialize"
+)
+
+// GetMetricIDForHistogramAgg returns a metric id (with sorted tag pairs) without the le tag (if is a histogram) and
+// strips histogram suffixes ("_bucket", "_sum". "_count")
+func GetMetricIDForHistogramAgg(metricID id.RawID) (id.RawID, bool) {
+	it := serialize.NewUncheckedMetricTagsIterator(serialize.NewTagSerializationLimits())
+	it.Reset(metricID)
+	leTagName := []byte("le")
+	nameTagName := []byte("__name__")
+
+	bucketSuffix := []byte("_bucket")
+	countSuffix := []byte("_count")
+	sumSuffix := []byte("_sum")
+
+	isHistogram := false
+
+	var idForHistogramAgg []byte
+	for it.Next() {
+		tagName, tagValue := it.Current()
+
+		if !bytes.Equal(tagName, leTagName) {
+			if bytes.Equal(tagName, nameTagName) {
```

Comment:
I'd suggest to verify name tag first since the id is sorted and `__name__` is before `le`, same comment to break the iteration if histogram is detected to save cpu cycles.